apps/server/src/modules/tool/context-external-tool/uc/context-external-tool.uc.spec.ts

```diff
@@ -98,6 +99,219 @@ describe('ContextExternalToolUc', () => {
 
 				expect(contextExternalToolValidationService.validate).toHaveBeenCalledWith(contextExternalTool);
 			});
+
+			it('should return the saved object', async () => {
+				const { contextExternalTool, userId } = setup();
+
+				const result = await uc.createContextExternalTool(userId, contextExternalTool);
+
+				expect(result).toEqual(contextExternalTool);
+			});
+		});
+
+		describe('when the user does not have permission', () => {
+			const setup = () => {
+				const userId: EntityId = 'userId';
+
+				const contextExternalTool: ContextExternalTool = contextExternalToolFactory.buildWithId({
+					displayName: 'Course',
+					contextRef: {
+						id: 'contextId',
+						type: ToolContextType.COURSE,
+					},
+				});
+
+				const error = new ForbiddenException();
+
+				contextExternalToolService.ensureContextPermissions.mockRejectedValue(error);
+
+				return {
+					contextExternalTool,
+					userId,
+					error,
+				};
+			};
+
+			it('should return forbidden and not save', async () => {
+				const { contextExternalTool, userId, error } = setup();
+
+				const func = () => uc.createContextExternalTool(userId, contextExternalTool);
+
+				await expect(func).rejects.toThrow(error);
+				expect(contextExternalToolService.saveContextExternalTool).not.toHaveBeenCalled();
```

Comment:
this is because you have to catch the error somehow and when you do a try/catch in a separate `it` then you have an empty catch body, but lint does not allow that and that makes this unnecessarily complicated